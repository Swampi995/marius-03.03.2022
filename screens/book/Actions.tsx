import { FC } from 'react';
import { StyleSheet } from 'react-native';
import { View, Button } from '../../components';
import { MARKET } from '../../services/ordersData';

interface ActionsProps {
    market: MARKET;
    setMarket: (market: MARKET) => void;
    killed: boolean;
    setKilled: (killed: boolean) => void;
}

const Actions: FC<ActionsProps> = ({ market, setMarket, killed, setKilled }) => {

    const toggleFeed = () => {
        setMarket(market === 'PI_XBTUSD' ? 'PI_ETHUSD' : 'PI_XBTUSD');
    }

    const killFeed = () => {
        setKilled(!killed);
    }

    return (
        <View style={styles.container}>
            <Button title={'Toggle Feed'} onPress={toggleFeed} />
            <Button type={'secondary'} title={'Kill Feed'} onPress={killFeed} />
        </View>
    );
}

export default Actions;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingTop: 16,
        paddingHorizontal: 30,
    },
});
