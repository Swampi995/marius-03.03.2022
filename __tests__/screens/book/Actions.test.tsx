import renderer from 'react-test-renderer';

import Actions from '../../../screens/book/Actions';

describe('<Actions />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Actions market='PI_XBTUSD' killed={false} setMarket={() => { }} setKilled={() => { }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
