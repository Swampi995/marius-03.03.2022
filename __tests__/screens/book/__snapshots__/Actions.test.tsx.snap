// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actions /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#121723",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 30,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5741d9",
        "borderRadius": 5,
        "minWidth": 150,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#bebfc5",
            "fontFamily": undefined,
          },
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      Toggle Feed
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#b91d1d",
        "borderRadius": 5,
        "minWidth": 150,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#bebfc5",
            "fontFamily": undefined,
          },
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      Kill Feed
    </Text>
  </View>
</View>
`;
